src/wallet.rs

```diff
@@ -80,8 +82,17 @@ impl Wallet {
         Ok(balance)
     }
 
-    pub fn validate_addr(&self, _addr: String) -> AddressValidationResult {
-        todo!()
+    pub fn validate_addr(&self, addr: String) -> AddressValidationResult {
+        let address = match Address::from_str(&addr) {
+            Ok(a) => a,
+            Err(_) => return AddressValidationResult::Invalid,
+        };
+
+        if address.network != self.wallet.lock().unwrap().network() {
+            AddressValidationResult::Invalid
+        } else {
+            AddressValidationResult::Valid
+        }
```

Comment:
```suggestion
        if address.network == self.wallet.lock().unwrap().network() {
            AddressValidationResult::Valid
        } else {
            AddressValidationResult::Invalid
        }
```

No need to negate the comparison imo.

Alternatively you could just:

```
pub fn validate_addr(&self, addr: String) -> AddressValidationResult {
        if let address = Ok(Address::from_str(&addr)) {
            if address.network == self.wallet.lock().unwrap().network() {
                AddressValidationResult::Valid
            }
        };

        AddressValidationResult::Invalid
}
```

But I also kinda like your proposal because it allows for returning a more specific `AddressValidationResult::WrongNetwork` as well, in case this will become interesting for the library consumer to know.

Reply:
I don't see the negation as a problem but it's exactly the same for me so I accepted your suggestion :)